components/slices/Process.vue

```diff
@@ -0,0 +1,349 @@
+        :key="`item-${rowIndex}`"
+        class="process-wrapper__row"
+      >
+        <div
+          v-for="(item, itemIndex) in itemsInARow"
+          :key="`item-${itemIndex}`"
+          class="process-wrapper__item"
+        >
+          <pix-image v-if="hasImage(item)" :field="item.item_image" />
+          <div>
+            <prismic-rich-text
+              class="process-wrapper-item__title"
+              :field="item.item_title"
+            />
+            <prismic-rich-text
+              class="process-wrapper-item__description"
+              :field="item.item_description"
+            />
+          </div>
+        </div>
+      </div>
+    </div>
+  </section>
+</template>
+
+<script>
+import {
+  DESKTOP_MIN_WIDTH,
+  EXTRA_LARGE_SCREEN_MIN_WIDTH,
+  LARGE_SCREEN_MIN_WIDTH,
+  TABLET_MIN_WIDTH,
+} from '~/config/breakpoints'
+import PixImage from '@/components/PixImage'
+
+export default {
+  name: 'ProcessSlice',
+  components: {
+    PixImage,
+  },
+  props: {
+    slice: {
+      type: Object,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      itemsByRow: [],
+    }
+  },
+  computed: {
+    background() {
+      return {
+        backgroundColor: `${this.slice.primary.process_background_color}`,
+      }
+    },
+    items() {
+      return this.slice.items
+    },
+    displayTitle() {
+      return this.slice.primary.process_display_title === 'Yes'
+    },
+    useSubtitle() {
+      return this.slice.primary.process_use_subtitle === 'Yes'
+    },
+    hasTitle() {
+      return (
+        this.slice.primary.process_title.length &&
+        this.slice.primary.process_title[0].text.length
+      )
+    },
+    hasSubtitle() {
+      return (
+        this.slice.primary.process_subtitle.length &&
+        this.slice.primary.process_subtitle[0].text.length
+      )
+    },
+    title() {
+      return this.slice.primary.process_title
+    },
+    subtitle() {
+      return this.slice.primary.process_subtitle
+    },
+  },
+  mounted() {
+    this.itemsByRow = this.splitItemsIntoRows()
+  },
+  methods: {
+    hasImage(item) {
+      return (
+        Object.keys(item.item_image).length &&
+        item.item_image.constructor === Object
+      )
+    },
+    findNbItemsInRow() {
+      if (process.client) {
+        const isTabletScreen = screen.width >= TABLET_MIN_WIDTH
+        const isLargeScreen = screen.width >= LARGE_SCREEN_MIN_WIDTH
+        const isExtraLargeScreen = screen.width >= EXTRA_LARGE_SCREEN_MIN_WIDTH
+        const isDesktopScreen = screen.width <= DESKTOP_MIN_WIDTH
+
+        if (this.items.length === 3) {
+          if (isLargeScreen) {
+            return 3
+          } else return 2
+        }
+        if (this.items.length === 4) {
+          if (isExtraLargeScreen) {
+            return 4
+          }
+          if (isLargeScreen) {
+            return 3
+          } else if (isDesktopScreen || isTabletScreen) {
+            return 2
+          }
+        }
```

Comment:
Je pense qu'on peut faire ça pour réduire la coxplexité cognitive

```suggestion
        if (this.items.length === 3 && isLargeScreen) {
          return 3
        }
        if (this.items.length === 4) {
          if (isExtraLargeScreen) {
            return 4
          }
          if (isLargeScreen) {
            return 3
          }
        }
        return 2
```
ou
```suggestion
		if(isLargeScreen) {
			if(this.items.length === 3) {
				return 3
			}
			if(this.items.length === 4) {
				return 4
			}
		}
        else if(isExtraLargeScreen) {
        	if(this.items.length === 4) {
        		return 4
        	}
        }
        else {
        	return 2
        }
```